### PR TITLE
Add new `geostoreDataApi` field to areas v2 model

### PR DESCRIPTION
**Pull request by ikas**

### CODE REVIEW

File: app/src/routes/api/v2/area.router.js [hunk 1]
```diff
@@ -337,7 +338,16 @@ class AreaRouterV2 {
             // Update status to saved if geostore already exists with status=saved
             area.status = isSaved ? 'saved' : 'pending';
             logger.info(`Updating area with id ${ctx.params.id} to status ${isSaved ? 'saved' : 'pending'}`);
+        } else if (ctx.request.body.geostore === null) {
+            area.geostore = null;
         }
+
+        if (ctx.request.body.geostoreDataApi) {
```
Comment (tiagojsag) [inline app/src/routes/api/v2/area.router.js]: something like

```
 if (geostoreDataApi in ctx.request.body {
            area.geostoreDataApi = ctx.request.body.geostoreDataApi;
        }
```

File: app/src/services/mail.service.js [hunk 2]
```diff
@@ -13,7 +13,6 @@ class MailService {
         } else {
             logger.info('Skipping init of Sparkpost client due to missing API key.');
         }
-        this.publicUrl = publicUrl;
```
Comment (tiagojsag) [inline app/src/services/mail.service.js:16]: hum.... is this intentional?

Comment (ikas) [reply app/src/services/mail.service.js:16]: it wasn't used :)

File: app/src/services/subscription.service.js [hunk 3]
```diff
@@ -30,6 +30,10 @@ class SubscriptionsService {
         // Build subscription params
         body.params = {};
 
+        if (area.geostoreDataApi) {
```
Comment (tiagojsag) [inline app/src/services/subscription.service.js:33]: if both geostoreDataApi  and geostore exist, this will lead to problems. I know that, by definition, they should be mutually exclusive, but if we rely on that, we should make it explicit - aka throw an error if both are set on create/update. Until then, having this sort of "half enforced assumption" will lead to confusing behavior.

File: app/src/services/subscription.service.js [hunk 4]
```diff
@@ -120,6 +124,14 @@ class SubscriptionsService {
                 _id: { $nin: [area.id, area.subscriptionId] }
             });
 
+            if (areas && areas.length > 0) area.status = 'saved';
```
Comment (tiagojsag) [inline app/src/services/subscription.service.js:127]: this seems unrelated to the scope of this PR

File: app/src/services/subscription.service.js [hunk 5]
```diff
@@ -120,6 +124,14 @@ class SubscriptionsService {
...
+        } else if (area.geostoreDataApi) {
+            const areas = await AreaModel.find({
+                status: 'saved',
+                geostore: area.geostoreDataApi,
```
Comment (tiagojsag) [inline app/src/services/subscription.service.js]: should it be "geostore" here?

**Review by tiagojsag — CHANGES_REQUESTED** [review]
see my comments throughout the code.

File: app/src/services/subscription.service.js (hunk 4 above)
Comment (ikas) [reply app/src/services/subscription.service.js:127]: We update the status of an area to 'saved' if at least one area with the same `geostore` already has status 'saved' in the DB. We want to keep this behavior for the newly added `geostoreDataApi` field, since the behavior should be exactly the same as with the `geostore` field

File: app/src/services/subscription.service.js (hunk 3 above)
Comment (ikas) [reply app/src/services/subscription.service.js:33]: Agreed, will do that :+1:
